docs/documentation/workflows.md

```diff
@@ -312,6 +312,21 @@ containing all the related exceptions.
 The exceptions can be handled
 by [`ErrorStatusHandler`](https://github.com/java-operator-sdk/java-operator-sdk/blob/14620657fcacc8254bb96b4293eded84c20ba685/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/ErrorStatusHandler.java)
 
+## Notes on Ordered Kubernetes Resource Cleanups
+
+In case there are Kubernetes Dependent Resources that are depend on each other, and are ordered -
+like when those represent some external resources and would be nice to have them created and deleted in order -
+on cleanup the resources will be deleted in reverse order. However, the workflow the implementation currently just
+calls delete on resource, in case it uses a finalizer, it should be checked by a `DeletePostCondition` if it
```

Comment:
This isn't really clear and should be better explained, as it's not clear why having a finalizer changes things and why a delete post condition is needed.